src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -115,27 +113,28 @@ void SignalDetectorClass::addData(const int8_t value)
 		*/
 	} else {
 		printOut();
-		SDC_PRINT(" addData oflow->");
-		SDC_PRINT(" mstart="); SDC_PRINT(mstart);
-		SDC_PRINT(" mend="); SDC_PRINT(mend);
+		DBG_PRINT(" addData oflow->");
+		DBG_PRINT(" mstart="); DBG_PRINT(mstart);
+		DBG_PRINT(" mend="); DBG_PRINT(mend);
 	}
-	SDC_PRINT(" val="); SDC_PRINT(value);
-	SDC_PRINT(" msglen="); SDC_PRINT(messageLen);
-	SDC_PRINT(" bytc="); SDC_PRINT(message.bytecount);
-	SDC_PRINT(" valc="); SDC_PRINT(message.valcount);
-	SDC_PRINT(" mTrunc="); SDC_PRINT(m_truncated);
-	SDC_PRINT(" state="); SDC_PRINT(state);
-	SDC_PRINT(" success="); 
-	SDC_PRINTLN(success);
-
+	DBG_PRINT(" val="); DBG_PRINT(value);
```

Comment:
Die Daten kommen durch diese Änderung jetzt an der seriellen Konsole raus denke ich und nicht mehr auf dem "Telnet" Port.
Das muss nicht schlecht sein, ist halt nicht mehr so wie vorher

Reply:
Diese Änderung ist ebenso ein Bestandteil davon, das man den Code in der Größe optimieren kann.

Die alte Variante wurde mit eincompiliert weil das Macro definiert wurde um letztendlich das gleiche nur an anderer Stelle zu machen. Somit habe ich es auf `DBG_PRINT` geändert weil ich der Meinung bin, 2x das selbe Makro nur als "KomfortableVariante" in der FW zu haben, so nehme ich lieber die Erweiterung und Performenceoptimierung für die uC als Priorität anstatt den Code "künstlich" den Flash oder RAM zu füllen damit.

Reply:
Die macros gehen nicht in die Firmware, die werden vom Präprozessor verarbeitet. Das hat am Ende keinen Einfluss auf die Größe, wie viele macros definiert wurden.

Das mit dem Debug anstelle SDC Print muss ja nicht schlecht sein, aber ich hatte es schon geahnt und jetzt auch verifizieret. Es gab einen Grund, wieso ich SDC_PRINT definiert hatte :)
SignalDecoder ist eine eigene Bibliothek. Die wird in das Hauptprogramm eingebunden.

Jetzt hast Du , einen Header in diese Bibliothek eingebunden, der nicht Bestandteil von dieser Bibliothek ist.
Das sollte man lieber erst nicht anfangen, besser ist, wenn die Bibliothek für sich funktioniert.

Der Travis Test läuft auch aus genau diesem Grund nicht, denn beim Compilieren der Bibliothek findet er output.h einfach nicht. :(

Reply:
> Die macros gehen nicht in die Firmware, die werden vom Präprozessor verarbeitet. Das hat am Ende keinen Einfluss auf die Größe, wie viele macros definiert wurden.

Doch es ist so. Fakten kommen nächste Woche.

> Das mit dem Debug anstelle SDC Print muss ja nicht schlecht sein, aber ich hatte es schon geahnt und jetzt auch verifizieret. Es gab einen Grund, wieso ich SDC_PRINT definiert hatte :)
SignalDecoder ist eine eigene Bibliothek. Die wird in das Hauptprogramm eingebunden.

Genau das SDC_Macro bringt aber Größe :D

> Jetzt hast Du , einen Header in diese Bibliothek eingebunden, der nicht Bestandteil von dieser Bibliothek ist.
Das sollte man lieber erst nicht anfangen, besser ist, wenn die Bibliothek für sich funktioniert.

Ja, verstehe ich, aber wieso definieren wir im Grunde fast immer das gleiche und quellen so den Code auf.